src/components/Modals/ApprovalFlowModal/ApprovalFlowModal.tsx

```diff
@@ -27,6 +29,14 @@ export default class ApprovalFlowModal extends React.PureComponent<Props> {
     onRescueItems(collection, items, contentHashes)
   }
 
+  handleConsumeSlots = () => {
+    const { metadata, onConsumeTPSlots } = this.props
+    const { collection, merkleTreeRoot, slots } = metadata as ApprovalFlowModalMetadata<ApprovalFlowModalView.CONSUME_TP_SLOTS>
+    this.setState({ didApproveConsumeSlots: true })
+    const thirdPartyId = extractThirdPartyId(collection.urn)
+    onConsumeTPSlots(thirdPartyId, slots, merkleTreeRoot)
+  }
+
   renderHash(hash: string) {
     return hash.slice(0, 6) + '...' + hash.slice(-6)
```

Comment:
Just because I saw you were using it, we could replace this with `src/lib/address.ts`'s  `shorten` ! either replacing this method entirely or the implementation

Reply:
I tried it out, but the `merkleTreeRoot` has 66 characters, so it's not an address, the `shorten` looks weird on it cause it leaves too man characters at the end.